longformer/longformer.py

```diff
@@ -204,8 +200,15 @@ def forward(
             # (bsz, seq_len, num_heads, extra attention count + 2*window+1)
             attn_weights = torch.cat((selected_attn_weights, attn_weights), dim=-1)
         if getattr(self, 'global_tokens', 0) > 0:
+            q_g = self.query_global(hidden_states)
+            k_g = self.key_global(hidden_states)
```

Comment:
you can save some memory and compute if you slice `hidden_states` here instead of slicing `k_g`, something like:
```
selected_k_g = self.key_global(hidden_states[:, self. global_tokens])
selected_attn_weights = torch.einsum('blhd,bshd->blhs', (q_g, selected_k_g))
``` 
 (not sure about the dimensions of `hidden_state`)

Reply:
Good point, addressed